### PR TITLE
add decimal column reader support

### PR DESCRIPTION
**Pull request by zuochunwei**
I have tested the reading of short decimal type column by query_benchmark

replace #244

### CODE REVIEW

File: velox/dwio/dwrf/reader/SelectiveShortDecimalColumnReader.h [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  SelectiveShortDecimalColumnReader(
+      const std::shared_ptr<const dwio::common::TypeWithId>& nodeType,
+      const TypePtr& dataType,
+      DwrfParams& params,
+      common::ScanSpec& scanSpec)
+      : SelectiveColumnReader(nodeType, params, scanSpec, nodeType->type) {
+    precision_ = dataType->asShortDecimal().precision();
+    scale_ = dataType->asShortDecimal().scale();
+
+    auto& stripe = params.stripeStreams();
```
Comment (rui-mo) [inline velox/dwio/dwrf/reader/SelectiveShortDecimalColumnReader.h:45]: const auto& ?

**Review by rui-mo — COMMENTED** [review]
Can we open a PR in Gluten to test this branch? This can help ensure the CI works.

**Comment by JkSelf** [comment]
Offline discussed with @zuochunwei to open one PR to test.

File: velox/dwio/dwrf/reader/SelectiveLongDecimalColumnReader.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  rawValues_ = values_->asMutable<char>();
+  getFlatValues<UnscaledLongDecimal, UnscaledLongDecimal>(
+      rows, result, type_, true);
+}
+
+const uint32_t SelectiveLongDecimalColumnReader::MAX_PRECISION_64;
+const uint32_t SelectiveLongDecimalColumnReader::MAX_PRECISION_128;
+
+const int64_t
+    SelectiveLongDecimalColumnReader::POWERS_OF_TEN[MAX_PRECISION_64 + 1] = {
```
[the remark this answers is not in the document]
Comment (zuochunwei) [reply velox/dwio/dwrf/reader/SelectiveLongDecimalColumnReader.cpp:123]: OK

**Comment by zuochunwei** [comment]
> Offline discussed with @zuochunwei to open one PR to test.

https://github.com/oap-project/gluten/pull/1613

File: velox/dwio/dwrf/reader/SelectiveShortDecimalColumnReader.h (hunk 1 above)
Comment (zuochunwei) [reply velox/dwio/dwrf/reader/SelectiveShortDecimalColumnReader.h]: done

**Comment by zuochunwei** [comment]
> Can we open a PR in Gluten to test this branch? This can help ensure the CI works.

https://github.com/oap-project/gluten/pull/1613

**Comment by JkSelf** [comment]
@zuochunwei Can you edit your velox repo and branch like this [ep/build-velox/src/get_velox.sh](https://github.com/oap-project/gluten/pull/1639/files#diff-768e423ceb1ee9c4ec02343cbac86e5357136f705e73b3f4daa8207fac38930a). And then CI can test your branch. Thanks.

**Pull request merged**